optuna/integration/allennlp/_train.py

```diff
@@ -0,0 +1,378 @@
+"""
+This file is ported from AllenNLP original implementation. (Under Apache 2.0 license)
+https://github.com/allenai/allennlp/blob/main/allennlp/commands/train.py
+"""
+
+import logging
+import os
+from os import PathLike
+from typing import List
+from typing import Optional
+from typing import Union
+
+from allennlp.commands.train import TrainModel
+from allennlp.common import logging as common_logging
+from allennlp.common import Params
+from allennlp.common import util as common_util
+from allennlp.common.checks import check_for_gpu
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.meta import Meta
+from allennlp.common.meta import META_NAME
```

Comment:
`allennlp.common.meta` was introduced in AllenNLP v2.5.0 (https://github.com/allenai/allennlp/releases/tag/v2.5.0).
To support older versions of AllenNLP, we may have to conditionally import them.